pkg/api/testing/fuzzer.go

```diff
@@ -429,6 +430,30 @@ func FuzzerFor(t *testing.T, version unversioned.GroupVersion, src rand.Source)
 				}
 			}
 		},
+		func(r *runtime.RawExtension, c fuzz.Continue) {
+			// Pick an arbitrary type and fuzz it
+			types := []runtime.Object{&api.Pod{}, &extensions.Deployment{}, &api.Service{}}
+			obj := types[c.Rand.Intn(len(types))]
+			c.Fuzz(obj)
+
+			// Find a codec for converting the object to raw bytes.  This is necessary for the
+			// api version and kind to be correctly set be serialization.
+			codec, err := testapi.GetCodecForObject(obj)
```

Comment:
The round trip tests do pass when fuzzing RawExtensions using this method.  Should they fail, or will they start to fail in the future?

FWIW, I don't think this piece is actually needed for the tests for this code to pass, but will be required by the code introducing Templates since they have a runtime.RawExtension.

Would you expand a bit on how I could get the codec here that is under test, or point me to an example of where that is done?  I spent > hour figuring out how to get this Codec + the encoding piece working here.  The runtime.Object fuzzer function just hard codes the bytes to an arbitrary string right now.  If there isn't a simple way to get the correct codec and this isn't going to work, WDYT of just doing the same thing here?